nova_vm/src/engine/bytecode/executable.rs

```diff
@@ -1171,84 +1171,76 @@ impl CompileEvaluation for ast::ArrayExpression<'_> {
-                    }
-                    ast::Argument::Super(x) => {
-                        x.compile(ctx);
-                        ctx.exe.add_instruction(Instruction::GetValue);
-                    }
-                    ast::Argument::ParenthesizedExpression(x) => {
-                        x.compile(ctx);
-                        if is_reference(&x.expression) {
-                            ctx.exe.add_instruction(Instruction::GetValue);
-                        }
-                    }
-                    ast::Argument::ComputedMemberExpression(x) => {
-                        x.compile(ctx);
-                        ctx.exe.add_instruction(Instruction::GetValue);
-                    }
-                    ast::Argument::StaticMemberExpression(x) => {
-                        x.compile(ctx);
-                        ctx.exe.add_instruction(Instruction::GetValue);
-                    }
-                    ast::Argument::PrivateFieldExpression(x) => {
-                        x.compile(ctx);
-                        ctx.exe.add_instruction(Instruction::GetValue);
-                    }
-                    ast::Argument::SpreadElement(_)
-                    | ast::Argument::JSXElement(_)
-                    | ast::Argument::JSXFragment(_)
-                    | ast::Argument::TSAsExpression(_)
-                    | ast::Argument::TSSatisfiesExpression(_)
-                    | ast::Argument::TSTypeAssertion(_)
-                    | ast::Argument::TSNonNullExpression(_)
-                    | ast::Argument::TSInstantiationExpression(_) => unreachable!(),
-                }
+
+            spread.argument.compile(ctx);
+            if is_reference(&spread.argument) {
+                ctx.exe.add_instruction(Instruction::GetValue);
+            }
+            ctx.exe.add_instruction(Instruction::GetIteratorSync);
+
+            let iteration_start = ctx.exe.get_jump_index_to_here();
+            let iteration_end = ctx
+                .exe
+                .add_instruction_with_jump_slot(Instruction::IteratorStepValue);
+            // result: value; stack: [num, ...args]
+            ctx.exe.add_instruction(Instruction::LoadStoreSwap);
+            // result: num; stack: [value, ...args]
+            ctx.exe.add_instruction(Instruction::Increment);
+            // result: num + 1; stack: [value, ...args]
+            ctx.exe.add_instruction(Instruction::Load);
+            // stack: [num + 1, value, ...args]
+            ctx.exe
+                .add_jump_instruction_to_index(Instruction::Jump, iteration_start);
+            ctx.exe.set_jump_target_here(iteration_end);
+        } else {
+            let expression = argument.to_expression();
+            expression.compile(ctx);
+            if is_reference(expression) {
+                ctx.exe.add_instruction(Instruction::GetValue);
+            }
+            if let Some(num_arguments) = known_num_arguments.as_mut() {
+                *num_arguments = num_arguments
```

Comment:
**issue (nitpicky):** If this reaches the max value, then we should already move to tracking the value on the stack as we can't tell the difference.

Reply:
Fixed.